File: freshmart/src/types/supplier.ts
// ============================================================================
// Types cho module Nhà cung cấp (Suppliers)
// Khớp 100% với DTOs của Backend ASP.NET Core
// ============================================================================

export interface Supplier {
  id: string;
  code: string;
  name: string;
  contactName?: string | null;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  taxCode?: string | null;
  bankAccount?: string | null;
  bankName?: string | null;
  isActive: boolean;
  notes?: string | null;
  productCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface SupplierFilterParams {
  search?: string;
  isActive?: boolean;
  page?: number;
  limit?: number;
}

export interface CreateSupplierRequest {
  name: string;
  contactName?: string;
  phone?: string;
  email?: string;
  address?: string;
  taxCode?: string;
  bankAccount?: string;
  bankName?: string;
  isActive?: boolean;
  notes?: string;
}

export interface UpdateSupplierRequest {
  name: string;
  contactName?: string;
  phone?: string;
  email?: string;
  address?: string;
  taxCode?: string;
  bankAccount?: string;
  bankName?: string;
  isActive?: boolean;
  notes?: string;
}

File: Models/Supplier.cs
namespace dotnet_backend_freshmart.Models
{
    public class Supplier
    {
        public Guid Id { get; set; }

        public string Code { get; set; } = string.Empty;

        public string Name { get; set; } = string.Empty;

        public string? ContactName { get; set; }

        public string? Phone { get; set; }

        public string? Email { get; set; }

        public string? Address { get; set; }

        public string? TaxCode { get; set; }

        public string? BankAccount { get; set; }

        public string? BankName { get; set; }

        public bool IsActive { get; set; } = true;

        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Quan hệ 1 Nhà cung cấp có nhiều Sản phẩm cung ứng
        public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    }
}
